ctru-rs/src/applets/swkbd.rs

```diff
@@ -448,27 +659,84 @@ impl SoftwareKeyboard {
     /// # }
     pub fn set_max_text_len(&mut self, len: u16) {
         self.state.max_text_len = len;
+
+        // Activate the specific validation rule for maximum length.
+        self.state.valid_input = ValidInput::FixedLen.into();
     }
+}
 
-    fn parse_swkbd_error(&self) -> Error {
-        match self.state.result {
-            ctru_sys::SWKBD_INVALID_INPUT => Error::InvalidParameters,
-            ctru_sys::SWKBD_OUTOFMEM => Error::OutOfMem,
-            ctru_sys::SWKBD_HOMEPRESSED => Error::HomePressed,
-            ctru_sys::SWKBD_RESETPRESSED => Error::ResetPressed,
-            ctru_sys::SWKBD_POWERPRESSED => Error::PowerPressed,
-            ctru_sys::SWKBD_PARENTAL_OK => Error::ParentalOk,
-            ctru_sys::SWKBD_PARENTAL_FAIL => Error::ParentalFail,
-            ctru_sys::SWKBD_BANNED_INPUT => Error::BannedInput,
-            _ => unreachable!(),
+impl ParentalLock {
+    /// Initialize a new configuration for the Parental Lock applet.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # let _runner = test_runner::GdbRunner::default();
+    /// # fn main() {
+    /// #
+    /// use ctru::applets::swkbd::ParentalLock;
+    ///
+    /// let parental_lock = ParentalLock::new();
+    /// #
+    /// # }
+    #[doc(alias = "swkbdInit")]
+    pub fn new() -> Self {
+        unsafe {
+            let mut state = Box::<SwkbdState>::default();
+            ctru_sys::swkbdInit(state.as_mut(), Kind::Normal.into(), 1, -1);
+            ctru_sys::swkbdSetFeatures(state.as_mut(), ctru_sys::SWKBD_PARENTAL);
+            Self { state }
+        }
+    }
+
+    /// Launch the Parental Lock applet based on the configuration and return a result depending on whether the operation was successful or not.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # let _runner = test_runner::GdbRunner::default();
+    /// # fn main() {
+    /// # use ctru::services::{apt::Apt, gfx::Gfx};
+    /// #
+    /// # let gfx = Gfx::new().unwrap();
+    /// # let apt = Apt::new().unwrap();
+    /// use ctru::applets::swkbd::{ParentalLock, Error};
+    ///
+    /// let mut parental_lock = ParentalLock::new();
+    ///
+    /// match parental_lock.launch(&apt, &gfx) {
+    ///     Ok(_) => println!("You can access parental-only features and settings."),
+    ///     Err(Error::ParentalFail) => println!("Is a kid trying to access this?"),
+    ///     Err(_) => println!("Something wrong happened during the parental lock prompt.")
+    /// }
+    /// #
+    /// # }
+    #[doc(alias = "swkbdInputText")]
+    #[allow(unused_variables)]
+    pub fn launch(&mut self, apt: &Apt, gfx: &Gfx) -> Result<(), Error> {
```

Comment:
Remove `#[allow(unused_variables)]` and use underscore prefixes on unused parameters